data_structures/stack.h

```diff
@@ -73,8 +69,20 @@ class stack {
         size = otherStack.size;
     }
 
+ private:
+    void deleteAllNodes() {
```

Comment:
would it be an improvement to use stackTop instead of curNode to maintain the top of the stack? 

```suggestion
void deleteAllNodes() {
   while (stackTop != nullptr) {
            auto tmpNode = stackTop;
            stackTop = stackTop->next;
            delete tmpNode;
        }
}
```

Reply:
You are absolutely right! Thank you for this remark.